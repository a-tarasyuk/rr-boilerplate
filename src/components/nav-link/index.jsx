import React from 'react';
import PropTypes from 'prop-types';
import { NavLink as Link } from 'react-router-dom';
import styles from './styles.css';

const NavLink = ({ children, href }) => (
  <Link
    to={href}
    exact
    className={styles['nav-link']}
    activeClassName={styles['nav-link_active']}
  >
    { children }
  </Link>
);

NavLink.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default NavLink;
