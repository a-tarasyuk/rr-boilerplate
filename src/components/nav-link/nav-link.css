.nav-link {
  font-size: 1.8em;
  color: #3572b0;
  display: inline-block;
  margin: 0 10px 0 0;
  text-decoration: none;

  &_active {
    text-decoration: underline;
  }

  &:hover {
    color: #ccc;
  }
}
