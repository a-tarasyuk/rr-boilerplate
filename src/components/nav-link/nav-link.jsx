import React from 'react';
import PropTypes from 'prop-types';
import { NavLink as Link } from 'react-router-dom';
import styles from './nav-link.css';

export const NavLink = ({ children, href }) => (
  <Link
    activeClassName={styles['nav-link_active']}
    className={styles['nav-link']}
    to={href}
    exact
  >
    { children }
  </Link>
);

NavLink.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default NavLink;
