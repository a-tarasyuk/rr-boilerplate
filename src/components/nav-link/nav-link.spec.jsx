import React from 'react';
import { shallow } from 'enzyme';
import { NavLink } from './nav-link';

describe('<NavLink />', () => {
  it('matches its snapshot', () => {
    const component = shallow(<NavLink href={ 'http://localhost' }>Link</NavLink>);
    expect(component).toMatchSnapshot();
  });
});
