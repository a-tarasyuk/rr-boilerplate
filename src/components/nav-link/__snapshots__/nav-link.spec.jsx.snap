// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> matches its snapshot 1`] = `
<NavLink
  activeClassName="nav-link_active"
  aria-current="page"
  className="nav-link"
  exact={true}
  to="http://localhost"
>
  Link
</NavLink>
`;
